Dockerfile

```
@@ -1,27 +1,57 @@
+# Builder stage
 FROM golang:1.23-alpine3.19 AS builder
+
+# Install necessary build dependencies
 RUN apk add --no-cache git make build-base
 
+# Set working directory
 WORKDIR /go/src/github.com/buildwithgrove/path
 
-# Copy only go.mod and go.sum first to cache dependency installation
+# Copy only go.mod and go.sum first to leverage Docker's build cache
 COPY go.mod go.sum ./
-RUN go mod download
+RUN go mod download && go mod verify
+
+# Set build flags for faster compilation
+ENV CGO_ENABLED=0
+ENV GOOS=linux
+ENV GOARCH=arm64
```

```suggestion
ENV GOARCH=amd64
```
Pretty sure this needs to be `amd64` to match the alpine image. The built image fails locally for me until I switch to `amd64`